### PR TITLE
LLM: ensure the result of daily arc perf test

### PR DESCRIPTION
**Pull request by WeiguangHan**
@hkvision Ensure the result of daily arc perf test

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -942,9 +985,10 @@ def run_transformer_autocast_bf16( repo_id,
...
-        df = pd.DataFrame(results, columns=['model', '1st token avg latency (ms)', '2+ avg latency (ms/token)', 'encoder time (ms)',
-                                            'input/output tokens', 'actual input/output tokens', 'num_beams', 'low_bit', 'cpu_embedding', 
-                                            'peak mem (GB)'])
+        if api!="transformer_int4_gpu":
+            df = pd.DataFrame(results, columns=['model', '1st token avg latency (ms)', '2+ avg latency (ms/token)', 'encoder time (ms)',
+                                                'input/output tokens', 'actual input/output tokens', 'num_beams', 'low_bit', 'cpu_embedding',
+                                                'peak mem (GB)'])
 
-        df.to_csv(f'{current_dir}/{api}-results-{today}.csv')
+            df.to_csv(csv_name)
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:993]: Does it work to keep the results and write to csv again for transformer_int4_gpu as well?

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 2]
```diff
@@ -933,6 +972,10 @@ def run_transformer_autocast_bf16( repo_id,
     
     import pandas as pd
     for api in conf.test_api:
+        global arc_perf_test_counter
+        arc_perf_test_counter = 0
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:976]: I don't think it needs be a global variable?

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 3]
```diff
@@ -432,12 +444,39 @@ def run_transformer_int4_gpu(repo_id,
...
             thread.join()
+
+            if result[in_out]:
+                new_row = []
+                first_token_latency = round(np.mean(result[in_out], axis=0)[0]*1000.0, 2)
+                rest_token_latency = round(np.mean(result[in_out], axis=0)[1]*1000.0, 2)
+                encoder_time = round(np.mean(result[in_out], axis=0)[2]*1000.0, 2)
+                input_output_tokens = in_out
+                actual_input_output_tokens = f'{int(np.mean(result[in_out], axis=0)[3])}' + f'-{int(np.mean(result[in_out], axis=0)[4])}'
+                cpu_embedding = cpu_embedding if 'win' in test_api else 'N/A'
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:455]: don't need this argument, it is always 'N/A'

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 4]
```diff
@@ -362,7 +363,9 @@ def run_transformer_int4_gpu(repo_id,
                              num_trials,
                              num_beams,
                              low_bit,
-                             batch_size):
+                             batch_size,
+                             test_api,
+                             cpu_embedding):
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:368]: no need these two arguments

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 5]
```diff
@@ -432,12 +444,39 @@ def run_transformer_int4_gpu(repo_id,
...
+                new_row.append(repo_id)
+                new_row.append(first_token_latency)
+                new_row.append(rest_token_latency)
+                new_row.append(encoder_time)
+                new_row.append(input_output_tokens)
+                new_row.append(actual_input_output_tokens)
+                new_row.append(num_beams)
+                new_row.append(low_bit)
+                new_row.append(cpu_embedding)
+                new_row.append(peak_mem)
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:469]: put all of them in one line

Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:469]: sure

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 3 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:455]: ok

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 2 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:976]: Will try to make it a local variable.

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 1 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: I think it may work.

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 6]
```diff
@@ -432,14 +442,28 @@ def run_transformer_int4_gpu(repo_id,
             thread = threading.Thread(target=run_model_in_thread, args=(model, in_out, tokenizer, result, warm_up, num_beams, input_ids, out_len, actual_in_len, num_trials))
             thread.start()
             thread.join()
+
+            if result[in_out]:
+                new_row = []
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:447]: no need this?

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 7]
```diff
@@ -405,6 +406,15 @@ def run_transformer_int4_gpu(repo_id,
 
     model = BenchmarkWrapper(model)
 
+    csv_data = []
+    try:
+        with open(csv_name, 'r') as csvfile:
+            csv_reader = csv.reader(csvfile)
+            csv_data = [row for row in csv_reader]
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:413]: why do we need to read the entire file? We just need to write?

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 6 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: sure

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 7 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: Have modified it.

File: python/llm/test/benchmark/concat_csv.py [hunk 8]
```diff
@@ -34,12 +34,17 @@ def main():
             csv_files.append(file_path)
     csv_files.sort()
 
-    df1 = pd.read_csv(csv_files[0], index_col=0)
-    df2 = pd.read_csv(csv_files[1], index_col=0)
-    merged_df = pd.concat([df1, df2], ignore_index=True)
-    merged_df.reset_index(drop=True, inplace=True)
+    if len(csv_files)>=2:
```
Comment (hkvision) [inline python/llm/test/benchmark/concat_csv.py:37]: may support csv files more than 2?

Comment (WeiguangHan) [reply python/llm/test/benchmark/concat_csv.py]: sure

**Review by hkvision — APPROVED** [review]

**Pull request merged**